tests/vm_test.go

```diff
@@ -297,32 +283,6 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			Expect(vmi.Annotations).To(HaveKeyWithValue("testannotation", "test"))
 		})
 
-		It("[test_id:3162]should ignore kubernetes and kubevirt annotations to VMI", func() {
```

Comment:
It would be great to have unit test replacing this one.

Reply:
This functionality was removed in your commit back then: https://github.com/kubevirt/kubevirt/commit/efa9e28374fb10713fa4eba7793e77e59e794c49, why then unit test is needed?

Reply:
A negative test to prevent anybody accidentally copying the VM annotations to VMI

Reply:
> A negative test to prevent anybody accidentally copying the VM annotations to VMI

This is too aggressive to see it in an e2e test.
In an unit test it makes sense. Is there such test?

Reply:
I see that we already have unit tests in `vm_test.go` to cover these test scenarios [1].

[1] https://github.com/kubevirt/kubevirt/pull/3110/files#diff-6a5a8576ebb4952ea056750d75d85e96d53eb62406defce2df9091dc9c0c0c2d

Reply:
>In an unit test it makes sense. Is there such test?

I still mean to replace this by unit test. We are missing this test.
Even if it is trivial it makes sense as people on the project changes and someone new to Kubevirt can report that annotation are not copied. New people can assume this was oversight, mistake and introduce it again...

Reply:
I've added a new unit test to cover it [1].

[1] https://github.com/kubevirt/kubevirt/compare/8e3aea1d9c81f537478dff8ef8cc960b87d90f96..523e8962c247d6f9df97859fc91369b9e56a899c

Reply:
@akalenyu Can you please add your opinion as well, that it can be over-testing like you said in the office today? Thanks!

Reply:
Yeah I am thinking it's really weird to test something that accidentally existed a few years ago, but I see you already have the test written so I wouldn't insist too much on this

Reply:
@xpivarc I think this PR contains lots of reviews, and multiple purposes. I'd like to stick to single purpose, because this PR only removes `RetryWithMetadataIfModified` function from `utils.go`.

1. @xpivarc If you agree, I'd like to keep the new unit test out of this PR, in a new PR, to get this PR merged that was already approved by @EdDev and I guess it looks good by @akalenyu.
2. If you disagree, I'll undo my changes in this PR and just move `RetryWithMetadataIfModified` function from `utils.go` to where it's used, and I'll keep this functional test that was removed.

Thanks!

Reply:
I am ok if we just remove `RetryWithMetadataIfModified`

Reply:
@xpivarc Sounds good. Let's do it this way, because currently both commits in this PR contain multiple purposes.